scripts/install_all_third_party_dependencies.py

```diff
@@ -0,0 +1,6 @@
+import subprocess
+
+from ts_utils.requirements import get_external_stub_requirements
+
+requirements = get_external_stub_requirements()
+subprocess.check_call(("pip", "install", *[str(requirement) for requirement in requirements]))
```

Comment:
nit: we have uv as a dependency in `requirements-tests.txt`

```suggestion
subprocess.check_call(("uv", "pip", "install", *map(str,requirements)))
```

Reply:
I initially considered it. But we still have a handful of places where we use `pip` directly instead of `uv`. And last I checked, `uv` doesn't tend to respect the activated environment (or lack thereof if contributor wants to install to system Python). Which feels like a speedbump to newer contributors or those using custom venvs.

From our contributing guidelines:
```bash
$ python3 -m venv .venv
$ source .venv/bin/activate
(.venv)$ pip install -U pip
(.venv)$ pip install -r requirements-tests.txt
```

The current places where `uv pip` is used in tests is where temporary venvs are created anyway.

Reply:
Hmm, I'm pretty sure it does respect the activated environment — if you have an example where it doesn't, that definitely sounds like a bug you should report! 😃 but you're right that some of the external requirements might not be PEP-517-compliant. So, probably best to stick with pip for now 👍

Reply:
(FWIW it _does_ seem like all the packages installed by this script are PEP-517 compliant and installable by uv)

Reply:
> Hmm, I'm pretty sure it does respect the activated environment — if you have an example where it doesn't, that definitely sounds like a bug you should report!

Looks like that may have just changed in `0.5.29`. This is what I used to get: https://github.com/astral-sh/uv/issues/1495#issuecomment-2629831612

> So, probably best to stick with pip for now 👍

That being said, I wouldn't mind having a follow-up that updates all `python -m pip`, `python3 -m pip`, and `pip` usages in docs and scripts for `uv pip`. (except for `pip install -r requirements-tests.txt` because that's a bit of a catch-22)

Reply:
> Looks like that may have just changed in `0.5.29`. This is what I used to get: [astral-sh/uv#1495 (comment)](https://github.com/astral-sh/uv/issues/1495#issuecomment-2629831612)

Oh, right. Yeah, it might not respect the activated environment for the "project API" commands like `uv sync`. It _should_ always respect it for any `uv pip` commands though, I _think_?